src/client/chessboard/custom-square-renderer.tsx

```diff
@@ -0,0 +1,39 @@
+import { CustomSquareProps } from "react-chessboard/dist/chessboard/types";
+import { ChessEngine } from "../../common/chess-engine";
+import { ReactElement, forwardRef } from "react";
+import { Circle, PieceSquare } from "./svg-components";
+
+export function getCustomSquareRenderer(
+    legalSquares: string[],
+    chess: ChessEngine,
+) {
+    return forwardRef((props: CustomSquareProps, ref) => {
```

Comment:
I'm not familiar with forwardRef - what's it for?

Reply:
So... the warning gotten without forwardRef is 
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```

I suspect this is due to the `ref` prop value being named as such, probably conflicting with existing JS or HTML attributes. Using forwardRef eliminates this problem.